ui/handler.go

```diff
@@ -121,6 +121,89 @@ func (h *internalHandler) handleNewRulesetRequest(w http.ResponseWriter, r *http
 
 }
 
+func (h *internalHandler) handleEditRulesetRequest(w http.ResponseWriter, r *http.Request) {
+	path := strings.TrimPrefix(r.URL.Path, "/rulesets/")
+	if path == "" {
+		writeError(w, r, nil, http.StatusNotFound)
+		return
+	}
+
+	nrr := &newRulesetRequest{}
+
+	err := json.NewDecoder(r.Body).Decode(nrr)
+	if err != nil {
+		writeError(w, r, err, http.StatusBadRequest)
+		return
+	}
+
+	// Get the existing entry
+	entry, err := h.service.Get(r.Context(), path, "")
+	if err != nil {
+		if err == store.ErrNotFound {
+			writeError(w, r, err, http.StatusNotFound)
+			return
+		}
+
+		writeError(w, r, err, http.StatusInternalServerError)
+		return
+	}
+
+	// Update the entry with the new rules
+	status, err := updateEntry(entry, nrr)
+	if err != nil {
+		writeError(w, r, err, status)
+		return
+	}
+
+	// Write the new entry back to the DB
+	result, err := h.service.Put(r.Context(), path, entry.Ruleset)
+	if err != nil {
+		writeError(w, r, err, http.StatusInternalServerError)
```

Comment:
I am afraid that if a ruleset is submitted without any changes the `Put` method will return `store.ErrNotModified` and here we will return `http.StatusInternalServerError`.

Reply:
OK. I'll add a test case and handle that appropriately.  :+1:

Reply:
DOne

Reply:
:up: @YassineE